src/helpers/manage-issue-due-dates.ts

```diff
@@ -0,0 +1,136 @@
+}
+
+export const manageIssueDueDates = async ({
+  warningThreshold = 7,
+  almostOverdueLabel = ALMOST_OVERDUE_ISSUE,
+  overdueLabel = OVERDUE_ISSUE,
+  priorityLabels = { priority_1: PRIORITY_1, priority_2: PRIORITY_2, priority_3: PRIORITY_3, priority_4: PRIORITY_4 }
+}: AddOverdueIssueLabel) => {
+  const openIssues: IssueList = await paginateAllOpenIssues(
+    [priorityLabels.priority_1, priorityLabels.priority_2, priorityLabels.priority_3, priorityLabels.priority_4].join()
+  );
+
+  const getPriorityLabel = (labels: IssueLabels, priorityLabels: PriorityLabelType) => {
+    if (!labels) return '';
+    // eslint-disable-next-line functional/no-let
+    let index = -1;
+    Object.values(priorityLabels).forEach(priority => {
+      if (index === -1) {
+        index = labels.findIndex(label => (label.name || label) === priority);
+      }
+    });
+    // eslint-disable-next-line functional/no-let
+    let priorityLabel = '';
+    if (index > -1) {
+      const label = labels[index];
+      if (typeof label === 'string') priorityLabel = label;
+      else priorityLabel = label.name;
+    }
+    return priorityLabel;
+  };
+
+  const addOverdueLabel = (priority: string, date_created: Date, issue_number: number, assignee: string) => {
+    const SLAGuidelines = {
+      [priorityLabels.priority_1]: 2,
+      [priorityLabels.priority_2]: 14,
+      [priorityLabels.priority_3]: 45,
+      [priorityLabels.priority_4]: 90
+    };
+
+    const daysSinceCreation = (Date.now() - date_created.getTime()) / 86400000;
+
+    // eslint-disable-next-line functional/no-let
+    let labelToAdd = '';
+
+    if (daysSinceCreation > SLAGuidelines[priority]) labelToAdd = overdueLabel;
+    else if (daysSinceCreation > SLAGuidelines[priority] - warningThreshold) labelToAdd = almostOverdueLabel;
+
+    if (labelToAdd.length && assignee.length) {
+      octokit.issues.createComment({
+        issue_number,
+        body: `@${assignee}, this issue assigned to you is now ${labelToAdd.toLowerCase()}`,
+        ...context.repo
+      });
+
+      return octokit.issues.addLabels({
+        labels: [labelToAdd],
+        issue_number,
+        ...context.repo
+      });
+    }
+  };
+
+  const addDueDateComment = async (deadline: number, createdDate: Date, issue_number: number, hasExistingComments: boolean) => {
+    const commentList: CommentList = hasExistingComments ? await paginateAllCommentsOnIssue(issue_number) : [];
+    // Create due date comment if there are no existing comments or the comment list does not contain a due date comment
+    if (
+      !hasExistingComments ||
+      commentList.findIndex((comment: SingleComment) => comment.body?.startsWith('This issue is due on')) === -1
+    ) {
+      const dueDate = new Date(createdDate.getTime() + deadline * 86400000);
+
+      octokit.issues.createComment({
+        issue_number,
+        body: `This issue is due on ${dueDate.toDateString()}`,
+        ...context.repo
+      });
+    }
+  };
```

Comment:
Could probably separate this function as well as `addOverdueLabel` into a separate file